src/api/core.jl

```diff
@@ -317,8 +317,8 @@ function _get_time_series_value(pv, t::TimeSlice, upd)
         timeout = _timeout(pv.value, t_start, t_end, a, b)
         _add_update(t, timeout, upd)
     end
-    t_end <= pv.value.indexes[1] && return NaN
-    t_start > pv.value.indexes[end] && !pv.value.ignore_year && return NaN
+    t_end <= pv.value.indexes[1] && throw(DomainError(pv, "Time index $t out of range!"))
+    t_start > pv.value.indexes[end] && !pv.value.ignore_year && throw(DomainError(pv, "Time index $t out of range!"))
```

Comment:
I believe we rely on time series returning NaN in those cases for a lot of things - I suggest we instead error out in SpineOpt whenever we find a NaN in the replacement value.

Reply:
My suggestion might also be more robust in case the NaN happens to be in the middle for some reason.